python/nano/src/bigdl/nano/utils/inference/pytorch/model_utils.py

```diff
@@ -32,6 +36,95 @@ def get_forward_args(model):
     return forward_args
 
 
+def get_forward_defaults(model):
+    '''
+    This function is to get all the defaults
+    It will return a list of default values
+    '''
+    forward_defaults = inspect.getfullargspec(model.forward).defaults
+    if isinstance(model, LightningModule):
+        if not isinstance(model, AcceleratedLightningModule):
+            # forward param list for compiled model
+            forward_defaults = get_forward_defaults(model.model)
+    return forward_defaults
+
+
+def get_forward_annotations(model):
+    '''
+    This function is to get all the annotations
+    It will return a dict of {args: annotations}
+    '''
+    forward_annotations = inspect.getfullargspec(model.forward).annotations
+    if isinstance(model, LightningModule):
+        if not isinstance(model, AcceleratedLightningModule):
+            # forward param list for compiled model
+            forward_annotations = get_forward_annotations(model.model)
+    return forward_annotations
+
+
+def get_tensor_args(model):
+    '''
+    This function will return all the parameters that (might) be a tensor type
+    It will return a list or tensor args name
+    '''
+    forward_args = get_forward_args(model)
+    forward_defaults = get_forward_defaults(model)
+    forward_annotations = get_forward_annotations(model)
+    tensor_args = []
+    if forward_defaults is None:
+        defaults_length = 0
+    else:
+        defaults_length = len(forward_defaults)
+    args_length = len(forward_args)
+    for i, arg in enumerate(forward_args):
+        # check if type hint is provided
+        if arg in forward_annotations:
+            if forward_annotations[arg] == torch.Tensor:
+                tensor_args.append(arg)
+            continue
+        # check if defaults is Tensor
+        default_args_start_from = args_length - defaults_length
+        if i >= default_args_start_from:
+            if type(forward_defaults[i - default_args_start_from]) == torch.Tensor:
+                tensor_args.append(arg)
+            continue
+        # nothing is provided, we assume it is a Tensor
+        tensor_args.append(arg)
+    return tensor_args
+
+
+def complement_input_sample(model, input_sample):
+    '''
+    This function will give a complemented input sample
+    Mainly using default value to complete.
+    '''
+    forward_args = get_forward_args(model)
+    forward_defaults = get_forward_defaults(model)
+    input_sample_length = 1
+    if isinstance(input_sample, Sequence):
```

Comment:
what if input_sample is a dict？
or now we only support sequence.

Reply:
Could we support dict as an input sample now? I did not find any ut for this.